### PR TITLE
Add an e2e test for out of process adapter

### PR DESCRIPTION
**Pull request by bianpengyuan**
Add a e2e test for out of process adapter with a testing out of process prometheus adapter.

#12506

### CODE REVIEW

File: tests/e2e/tests/mixer/mixer_test.go [hunk 1]
```diff
@@ -463,16 +474,31 @@ func errorf(t *testing.T, format string, args ...interface{}) {
...
+	}
+
+	defer func() {
+		if err := deleteMixerRule(oopPromConfigs); err != nil {
+			t.Logf("could not clear out of process config: %v", err)
+		}
+	}()
+	allowRuleSync()
+
+	checkMetricReport(t, "istio_request_count_oop", destLabel, fqdn("productpage"))
```
Comment (kyessenov) [inline tests/e2e/tests/mixer/mixer_test.go]: Do I understand correctly that this measure the number of OOP calls? If so, I'd like to see some functional check as well. Maybe call some application endpoint and measure application metric?

**Comment by kyessenov** [comment]
LG. I do wonder what the plan is about migration to the new testing framework, but that's a bigger question.

File: tests/e2e/tests/mixer/mixer_test.go (hunk 1 above)
Comment (bianpengyuan) [reply tests/e2e/tests/mixer/mixer_test.go]: No, this is a replication of default request count metric: https://github.com/istio/istio/pull/12577/files#diff-42f57e2875ff647fa7593c2fc4675c53R44. So this is actually a functional check.

Comment (kyessenov) [reply tests/e2e/tests/mixer/mixer_test.go]: OK thanks for clarifying.

**Comment by bianpengyuan** [comment]
@utka May I get your approval on this? Thanks!

**Comment by kyessenov** [comment]
Master in unblocked. You just need to assign it to one of us. Bot lost capability to assign people at some point.

**Comment by kyessenov** [comment]
E.g. you need to assign it to @utka

**Pull request merged**